File: Chess/src/Pieces/King.cpp
#include <cstdlib>
#include "King.hpp"
#include "../VectorUtils.hpp"

bool King::canMove(sf::Vector2f position) {
	sf::Vector2f diff = abs(this->getBoardPosition() - position);
	return !(diff.x > 1 || diff.y > 1);
}

int King::getTextureIndex() {
	return 0;
}

File: Chess.NET/Pieces/King.cs
using System;
using System.Collections.Generic;
using System.Linq;
using SFML.System;

namespace Chess.Pieces {
	public class King : Piece {
		public King(Vector2f position, Team team) : base(position, team) { }

		public override int TextureIndex => 0;

		public override bool CanMove(Vector2f to) {
			Vector2f diff = (BoardPosition - to).Abs();
			return !(diff.X > 1 || diff.Y > 1);
		}
	}
}
